### PR TITLE
Network: add listable

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
Signed-off-by: Pierre-Emmanuel Jacquier <pierre-emmanuel.jacquier@epitech.eu>

### CODE REVIEW

File: networks.go [hunk 1]
```diff
@@ -1,127 +1,39 @@
...
+		CanUseForDeploy:   &network.CanUseForDeploy,
+		DisplayNetwork:    &network.DisplayNetwork,
+		DomainID:          network.DomainID,
+		ForVpc:            network.VpcID,
+		ID:                network.ID,
+		PhysicalNetworkID: network.PhysicalNetworkID,
+		ProjectID:         network.ProjectID,
+		RestartRequired:   &network.RestartRequired,
+		SpecifyRanges:     &network.SpecifyIPRanges,
+		//Tags: network.Tags
```
Comment (greut) [inline networks.go]: add TODO or XXX so it appears when you search for them

File: networks_test.go [hunk 2]
```diff
@@ -76,3 +76,298 @@ func TestCreateNetworkOnBeforeSend(t *testing.T) {
...
+	cs := NewClient(ts.URL, "KEY", "SECRET")
+
+	network := new(Network)
+	networks, err := cs.List(network)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(networks) != 5 {
+		t.Errorf("Four zones were expected, got %d", len(networks))
```
Comment (greut) [inline networks_test.go]: four ?

File: networks_test.go [hunk 3]
```diff
@@ -76,3 +76,298 @@ func TestCreateNetworkOnBeforeSend(t *testing.T) {
...
+	}
+
+	if networks[0].(*Network).ID != "939a8c40-75b5-4a82-9d7e-f8813a26cf7c" {
+		t.Errorf("Expected ID to be third, got %#v", networks[0])
```
Comment (greut) [inline networks_test.go]: third?

Comment (pierre-emmanuelJ) [reply networks_test.go]: My bad (copy, paste) !

File: networks_test.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply networks_test.go]: thanks !

**Comment by pierre-emmanuelJ** [comment]
Why again ?
image

**Comment by greut** [comment]
@pierre-emmanuelJ splitting files doesn't affect the overall coverage. https://blog.golang.org/cover

**Comment by pierre-emmanuelJ** [comment]
i have added more test: https://github.com/exoscale/egoscale/pull/80/commits/6e7ee416ebd814e706d9f0e0301396ec2b1db64f

**Comment by greut** [comment]
@pierre-emmanuelJ use `go tool cover -html coverage.out` to see what's not tested

**Comment by pierre-emmanuelJ** [comment]
In all file using this way to check if cast doesn't fail coverage can't calculate it !

image

**Comment by greut** [comment]
@pierre-emmanuelJ right, but you can still do it.

**Comment by greut** [comment]
Otherwise, this looks good to me. I'll merge.

**Comment by pierre-emmanuelJ** [comment]
i'll open PR to write some tests on all these functions

**Pull request closed without merging**